cypress/integration/mocked/okta_register.spec.ts

```diff
@@ -0,0 +1,205 @@
+          userId: 'xxx',
+          displayName: 'xxx',
+          email: 'xxx',
+          emailValidated: false,
+          redirect: {
+            url: 'xxx',
+          },
+        },
+        '/auth/redirect',
+      );
+
+      cy.visit('/register?useOkta=true');
+
+      cy.url().should(
+        'eq',
+        'http://localhost:8861/register?returnUrl=https%3A%2F%2Fm.code.dev-theguardian.com&useOkta=true',
+      );
+      cy.getCookie('GU_U').should('not.exist');
+      cy.getCookie('SC_GU_LA').should('not.exist');
+      cy.getCookie('SC_GU_U').should('not.exist');
+    });
+  });
+
+  context('Signed in user posts to /register', () => {
+    beforeEach(() => {
+      cy.mockPurge();
+      cy.clearCookies();
+      //we visit the healthcheck page to make sure the cookies are clearer from the browser
+      cy.visit('/healthcheck');
+    });
+    it('should redirect to manage.theguardian.com if the sid Okta session cookie is valid', () => {
+      cy.mockPattern(
+        200,
+        {
+          id: 'test',
+          login: 'user@example.com',
+          userId: 'userId',
+          status: 'ACTIVE',
+          expiresAt: '2016-01-03T09:13:17.000Z',
+          lastPasswordVerification: '2016-01-03T07:02:00.000Z',
+          lastFactorVerification: null,
+          amr: ['pwd'],
+          idp: {
+            id: '01a2bcdef3GHIJKLMNOP',
+            type: 'OKTA',
+          },
+          mfaActive: true,
+        },
+        '/api/v1/sessions/the_sid_cookie',
+      );
+
+      cy.intercept('https://manage.theguardian.com').as('manage');
+
+      cy.mockPattern(204, {}, '/api/v1/users/userId/sessions');
+
+      cy.visit('/register?useOkta=true');
+
+      cy.setCookie('sid', `the_sid_cookie`);
+
+      cy.get('input[name="email"]').type('example@example.com');
+      cy.mockNext(200, {
+        userType: 'new',
+      });
+      cy.mockNext(200, {
+        status: 'success',
+        errors: [],
+      });
+      cy.get('[data-cy=register-button]').click();
+
+      cy.url().should(
+        'eq',
+        'https://profile.theguardian.com/signin?returnUrl=https%3A%2F%2Fmanage.theguardian.com%2F',
+      );
+    });
+
+    it('should redirect to /register if the sid Okta session cookie is set, but invalid', () => {
+      cy.mockPattern(404, '/api/v1/sessions/the_sid_cookie');
+
+      cy.mockPattern(204, {}, '/api/v1/users/userId/sessions');
+
+      cy.visit('/register?useOkta=true');
+
+      cy.setCookie('sid', `the_sid_cookie`);
+
+      cy.get('input[name="email"]').type('example@example.com');
+      //mock the response from register
+      cy.mockNext(200, {
+        userType: 'new',
+      });
+      cy.mockNext(200, {
+        status: 'success',
+        errors: [],
+      });
+
+      cy.get('[data-cy=register-button]').click();
+
+      cy.intercept('/register*').as('register');
+
+      //we redirect to /register and unset the sid cookie, so we need to wait till the redirect has happened
+      cy.wait('@register');
```

Comment:
IIRC, `cy.url().should()` already has retry logic in my experience, so you shouldn't need to wait for a redirect. Any reason you particularly need to use `cy.wait()` here, maybe `cy.location()` doesn't have the retry logic (but I think cypress assertions generally do).

Reply:
Actually I'm looking at the docs and this aliasing seems pretty cool, maybe we should be doing that instead of just expecting cypress to keep retrying the assertion. I'm easy either way.

https://docs.cypress.io/api/commands/intercept#Aliasing-an-intercepted-route